Graphics/MusicWheelItem Song NormalPart/get_lamp.lua

```diff
@@ -0,0 +1,86 @@
+local args = {...}
+local graphic = args[1]
+local pn = args[2]
+
+LampColors = {
+    [0]={1,1,1,0},
+    color "#ffffff",
+    color "#34bfff",
+    color "#48ff1d",
+    color "#fff568",
+    color "#bfeaff"
+}
+
+
+local function getLamp(self, song)
+    local best_lamp = 0
+
+    if song then
+        -- Default to the machine profile if not found
+        if PROFILEMAN:IsPersistentProfile(pn) then 
+          profile = PROFILEMAN:GetProfile(pn)
+        else 
+          profile = PROFILEMAN:GetMachineProfile()
+        end
+                
+        local diff = GAMESTATE:GetCurrentSteps(pn):GetDifficulty()
+        local st = GAMESTATE:GetCurrentStyle():GetStepsType()
+        local steps = song:GetOneSteps(st, diff)
+        if steps then
+            high_score_list = profile:GetHighScoreListIfExists(song, steps)
+            
+            -- Unplayed song (or played, but quit out)
+            if high_score_list == nil then
+                return 0
+            end
+            
+            for score in ivalues(high_score_list:GetHighScores()) do
+                if score then
+
+                    -- Misses are slightly more complicated to check
+                    local misses = score:GetTapNoteScore("TapNoteScore_Miss") +
+                                   score:GetTapNoteScore("TapNoteScore_CheckpointMiss") +
+                                   score:GetTapNoteScore("TapNoteScore_HitMine") +
+                                   score:GetHoldNoteScore("HoldNoteScore_LetGo")
+
+                    local note_counts = {
+                        misses,
+                        score:GetTapNoteScore("TapNoteScore_W5"),
+                        score:GetTapNoteScore("TapNoteScore_W4"),
+                        score:GetTapNoteScore("TapNoteScore_W3"),
+                        score:GetTapNoteScore("TapNoteScore_W2"),
+                        score:GetTapNoteScore("TapNoteScore_W1")
+                    }
+                
+
+                    current_lamp = 0
+                    -- 0 is not an fc, 1 is no misses, 2 is a good fc, 3 great fc, 4 pfc, 5 mfc
+
+                    for i, v in ipairs(note_counts) do
+                        if v == 0 then
+                            current_lamp = current_lamp + 1
+                        else
+                            break
+                        end
+                    end
+                    if current_lamp > best_lamp then
+                        best_lamp = current_lamp
+                    end
+                end -- if score then
+            end -- for score in ivalues
+        end -- if steps then
+    end -- if song then
+    return best_lamp
+end
+
+
+return LoadActor(graphic)..{
+    SetCommand=function(self, param)
+        self:diffuse(LampColors[getLamp(self, param.Song)])
+        self:align(1, 0.5)
+        -- Align P2's lamps to the right of the grade.
+        if pn == PLAYER_2 then
+            self:x(67)
```

Comment:
How does this work on other resolutions (4:3, 16:9, 21:9)?

Reply:
Poorly, see the followup screenshots, still tweaking it--but if you've got any suggestions on how to align it I'm all ears 🙂